### PR TITLE
GSoC 2017 blog post : data visualization using daru-view 

### PR DESCRIPTION
**Pull request by Shekharrajak**
- [x] Add more examples to show the usage.
- [x] Add images for the output.
- [x] Usage in web application

### CODE REVIEW

**Comment by Shekharrajak** [comment]
I am improving the blog. If you have any suggestions then let me know.  @v0dro @lokeshh @zverok.

**Comment by zverok** [comment]
Reviewing it currently but got distracted, will finish in an ~hour

**Comment by Shekharrajak** [comment]
Thanks @zverok , I will follow this.

**Comment by v0dro** [comment]
@Shekharrajak please finish with Victor's comments soon as we need to submit your blog post to the GSOC committee. Its ok if it stays a PR but finish the content.

**Comment by Shekharrajak** [comment]
@v0dro , I will finish it by tonight.

**Comment by Shekharrajak** [comment]
Due to power cut, I was not able to finish the blog. I will do it today.

**Comment by v0dro** [comment]
Alright. Can you add this blog post to your personal blog in http://shekharrajak.github.io/gsoc_2017_posts/ as well? GSOC reviewers can also quickly find it that way.

File: source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+visualisation? Then you must try [`daru-view`](https://github.com/Shekharrajak/daru-view), a plugin gem for [`daru`](https://github.com/SciRuby/daru).
+
+## What makes daru-view different ?
+
+* `daru-view` is desinged for interactive plotting of charts and tables.It
+provide different plotting tools like Nyaplot, HighCharts, GoogleCharts,
+DataTable. So you don't have to write any JavaScript code from these sites
+and no need to shift to other language to get charts.
+
+* It can work with **any ruby web application framework like Rails/Sinatra/Nanoc**. If you want to try few examples then please look into the
```
Comment (v0dro) [inline source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown]: This should work with hanami too right? Add that too.

File: source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+`daru-view/spec/dummy_*` examples of Rails, Sinatra, Nanoc web applications.
```
Comment (v0dro) [inline source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown]: Add link here (to the folder name).

File: source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+
+* Now Ruby developers are using IRuby notebook for few lines of code.
```
Comment (v0dro) [inline source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown]: Rephrase this as `Now Ruby developers are using IRuby notebook for interactive programming`.

File: source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+Daru is doing pretty good work as the data analysis & manipulation in IRuby notebook as well as backend part of web application. Ruby web application frameworks like Ruby on Rails, Sinatra, Nanoc are popular frameworks. So if Ruby developers get the gem like daru which can do data analysis
+and visualization work in applications, then there is no need of shifting to another language or usage of other gem.
+
+My project for <abbr title="Google Summer of Code 2017"> GSoC 2017</abbr> was to "make Daru more ready for integration with modern Web framework" in terms of visualization.
+
+To improve in terms of viewing data,
+[daru-view](https://github.com/Shekharrajak/daru-view), a plugin gem for
+[daru](https://github.com/SciRuby/daru) is created. [daru-view](https://github.com/Shekharrajak/daru-view) is for easy and interactive plotting in web application & IRuby notebook. It can work in frameworks like Rails, Sinatra, Nanoc and hopefully in others too.
+
+If you just want to see the features of daru-view, you can check these
```
Comment (v0dro) [inline source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown]: Rephrase this as : `To see a quick overview of daru-view's features, have a look at these examples:`

File: source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+examples :
+
+* [IRuby Notebook Examples](http://nbviewer.jupyter.org/github/shekharrajak/daru-view/tree/master/spec/dummy_iruby/)
+
+* [daru io and daru-view usage in Rails app](https://github.com/Shekharrajak/daru_examples_io_view_rails)
+
+* [Readme of daru-view](https://github.com/Shekharrajak/daru-view)
```
Comment (v0dro) [inline source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown]: README should be all caps and linking to the README.md file not the full repo. I would put the README as the

File: source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+)
+
+```
+
+
+Note: If you have already loaded the dependent JS files for the library then you can use `adpater: :googlecharts` in your Plot initialization.
+
+
+{% img https://github.com/Shekharrajak/gsoc_2017_blog/blob/master/img/googlechart_geo1.png?raw=true 'Baisc Geo chart using GoogleCharts' 'Baisc Geo chart using GoogleCharts' %}
+
```
Comment (v0dro) [inline source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown]: Maybe add small examples of each plotting library supported?

File: source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown [hunk 7]
```diff
@@ -0,0 +1,284 @@
...
+## Design of daru-view
+
+[daru-view](https://github.com/Shekharrajak/daru-view), currently using
+[Nyaplot](https://github.com/SciRuby/nyaplot), [HighCharts](https://www.highcharts.com/), [GoogleCharts](https://developers.google.com/chart/interactive/docs/gallery) for plotting the charts. It is also
+generating tables using [DataTables](https://datatables.net/) and [GoogleCharts](https://developers.google.com/chart/interactive/docs/gallery) with pagination, search and various features.
+
+
+### Design Pattern in daru-view
+
```
Comment (v0dro) [inline source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown]: Write a paragraph on how and where exactly does daru-view use these patterns so readers can find out themselves easily (with links).

File: source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown [hunk 8]
```diff
@@ -0,0 +1,284 @@
...
+To use it, daru-view uses [https://github.com/Shekharrajak/data_tables](https://github.com/Shekharrajak/data_tables) gem. [Note: the gem name will be changed in near future]
+
+It basically uses the HTML table code and add features that user want. So internally HTML table code of daru DataFrame and daru Vector is passed
+as data source parameter.
+
+
+## Future Work
+
+daru-view will be more powerful and simple in near future. Developers can add
+more libraris in daru-view easily, if required.
```
Comment (v0dro) [inline source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown]: Typo: libraries

Comment (v0dro) [reply source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown]: Is there any documentation that explicitly states HOW new libraries can be added? If not, add it in the CONTRIBUTING.md file or wiki and link it here.

File: source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown [hunk 9]
```diff
@@ -0,0 +1,284 @@
...
+
+If you are using daru gem for analysing the data and want to visualise it, then it will be good if you have data-visualisation within daru and can plot it directly using DataFrame/Vector objects of daru.
+
+daru-view will be helpful in plotting charts and tables directly from the Daru::DataFrame and Daru::Vector . daru-view using nyaplot, highcharts , google charts right now to plot the chart. So user can set the plotting library and get the chart accordingly.
+
+Most of the plotting libraries doesn't provide the features of plotting charts in iruby notebook. They are defined only for web applications (mostly for Rails). But daru-view can plot charts in any ruby web application as well as iruby notebook.
+
+## Acknowledgements
+
+I would like to thank to my mentors [Sameer Deshmukh](https://github.com/v0dro)
```
Comment (v0dro) [inline source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown]: Also add a note about the Ruby Science Foundation (with link) and Google Summer of Code 2017.

File: source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown [hunk 10]
```diff
@@ -0,0 +1,290 @@
...
+          ['2004',  1000],
+          ['2005',  1170],
+          ['2006',  660],
+          ['2007',  1030]
+]
+df_sale_exp = Daru::DataFrame.rows(data_rows)
+df_sale_exp.vectors = idx
+```
+
+do some manupulation using `daru` if you want to do. Now time to plot it:
```
Comment (v0dro) [inline source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown]: Remove this line and add a comment at the bottom of the above code snippet:
``` ruby
# perform data manipulations....
```

Comment (v0dro) [reply source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown]: Just keep the `Now time to plot it:` part.

File: source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown [hunk 11]
```diff
@@ -0,0 +1,290 @@
...
+
+```ruby
+line_basic_chart = Daru::View::Plot.new(df_sale_exp)
+line_basic_chart.chart
+```
+
+This will return the chart object we created using GoogleCharts.
+In IRuby notebook, you will like this:
```
Comment (v0dro) [inline source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown]: Typo: `you will see this`

File: source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown (hunk 9 above)
Comment (Shekharrajak) [reply source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown]: Done.

File: source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown (hunk 7 above)
Comment (Shekharrajak) [reply source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown]: Done

File: source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown (hunk 6 above)
Comment (Shekharrajak) [reply source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown]: I have added examples for Highcharts and Nyaplot as well. May be I should create different blog for details examples.

File: source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown (hunk 10 above)
Comment (Shekharrajak) [reply source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown]: Done.

File: source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown (hunk 11 above)
Comment (Shekharrajak) [reply source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown]: Done

File: source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown [hunk 12]
```diff
@@ -0,0 +1,461 @@
...
+load the required js files in your webpage or IRuby notebook.
+
+```ruby
+require 'daru/view'
+Daru::View.plotting_library = :googlecharts
+```
+
+Let's create a DataFrame :
+
+```
```
Comment (zverok) [inline source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown]: "ruby" word forgotten ;)

Comment (Shekharrajak) [reply source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown]: Done.

File: source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown (hunk 8 above)
Comment (Shekharrajak) [reply source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown]: Done.

**Comment by Shekharrajak** [comment]
Thanks @zverok @v0dro for reviewing and patience. I will add some details part on the wiki page of `daru-view` and link them here.

File: source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown [hunk 13]
```diff
@@ -410,7 +431,7 @@ as data source parameter.
 ## Future Work
 
 daru-view will be more powerful and simple in near future. Developers can add
-more libraris in daru-view easily, if required. To add library follow the setups given in 
+more libraris in daru-view easily, if required. To add library follow the setups given in
```
Comment (v0dro) [inline source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown]: Typo: libraries

Comment (v0dro) [reply source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown]: Just run the whole thing through a spell checker?

Comment (Shekharrajak) [reply source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown]: Yes, spell checking done. Hopefully now it is okay.

**Comment by v0dro** [comment]
@mohawkjohn can you merge this ASAP? Its been a long time.
